File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/2473-max-sum-of-a-pair-with-equal-sum-of-digits.c
int maximumSum (int* nums, int numsSize) {
    int mp [82];
    memset (mp, -1, sizeof (mp));
    int ans = -1;
    for (int i = 0; i < numsSize; i++) {
        int sumDigits = 0, temp = nums [i];
        while (temp) sumDigits += temp % 10, temp /= 10;
        if (mp [sumDigits] != -1)
            ans = fmax (ans, nums [i] + mp [sumDigits]);
        mp [sumDigits] = fmax (mp [sumDigits], nums [i]);
    }
    return ans;
}

File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/2473-max-sum-of-a-pair-with-equal-sum-of-digits.py
class Solution:
    def maximumSum (self, nums : List [int]) -> int:
        mp = defaultdict (list)
        
        for num in nums:
            digit_sum = sum (int (d) for d in str (num))
            heapq.heappush (mp [digit_sum], -num)
        
        ans = -1
        for values in mp.values ():
            if len (values) > 1:
                max1 = -heapq.heappop (values)
                max2 = -heapq.heappop (values)
                ans = max (ans, max1 + max2)
                
        return ans